pkg/connection/store/kubernetes/client.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+// Error strings.
+const (
+	errLoadKubeconfig   = "cannot load kubeconfig"
+	errBuildRestConfig  = "cannot build rest config kubeconfig"
+	errCreateClient     = "cannot create Kubernetes client"
+	errNoCurrentContext = "currentContext not set in kubeconfig"
+
+	errFmtNoClusterForContext = "cluster for currentContext %q not found"
+)
+
+// Note(turkenh): We should probably move this function to a more common package
+// and expose to be used by K8s based providers, i.e. provider-helm and
+// provider-kubernetes.
```

Comment:
Are you thinking you'll do this in a separate PR?

Reply:
I found out that there is a `clientcmd.RESTConfigFromKubeConfig` function with which I could remove this file completely. Remaining seems to be to small to have a common method:

```go
	config, err := clientcmd.RESTConfigFromKubeConfig(kfg)
	if err != nil {
		return nil, errors.Wrap(err, errBuildRestConfig)
	}
	remote, err := client.New(config, client.Options{})
	if err != nil {
		return nil, errors.Wrap(err, errCreateClient)
	}
```

There would still be a need for some common package to support gke auth (and others) but I would prefer to leave that out of the scope of this PR.